tools-local/tasks/mobile.tasks/WasmBundleTask/WasmBundleTask.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Net;
+using System.Reflection;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+public class WasmBundleTask : Task
+{
+    [Required]
+    public string? InputDirectory { get; set; }
+
+    [Required]
+    public string? FileName { get; set; } 
+
+    private (byte[] json_bytes, MemoryStream stream) EnumerateData() 
+    {
+        var indices = new List<object[]>();
+        var stream = new MemoryStream();
+        
+        var directoryInfo = new DirectoryInfo(InputDirectory!);
+        
+        foreach (var entry in directoryInfo.EnumerateFiles("*", SearchOption.AllDirectories)) 
+        {
+
+            var relativePath = Path.GetRelativePath(InputDirectory!, entry.FullName);
+            indices.Add(new object[] { relativePath, entry.Length });
+
+            using (var readStream = entry.OpenRead())
+                readStream.CopyTo(stream);
+        }
+        
+        stream.Position = 0;
+        var jsonBytes = JsonSerializer.SerializeToUtf8Bytes(indices);
+        
+        return (jsonBytes, stream);
+    }
+
+    public override bool Execute()
+    {
+        (byte[] json_bytes, MemoryStream stream) data;
+
+        if (InputDirectory == null) 
+        {
+            throw new ArgumentException("Input directory doesn't exist.");
+        }
+
+        data = EnumerateData();
+        
+        if (FileName == null) 
```

Comment:
Same idea here, since `FileName` is Required now, msbuild will ensure that it is not null. So, here do:

```
if (!File.Exists(FileName))
{
   Log.LogError($"Input filename '${FileName}' does not exist.");
   return false;
}
```

Logging the error like this ensures that a neat error message gets surfaced to the list of errors, and to the IDE.
And returning false here indicates to msbuild that the task failed.
Throwing an exception also does the same thing, except that you get exception trace, and message etc, which should happen only in case of something we didn't expect.